### PR TITLE
Fixes the 50% time dilation on every server by removing demos, an admin only tool that is the laggiest thing on the planet for incredibly little gain.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request
removes demos from the game, saving the universe and ending lag

## Why It's Good For The Game

yMTh

demos are a crappy admin tool that doesn't work right, is unreliable as shit, and is super fucking laggy

we do not need to maintain it in the code and continue lagging the servers

no, we will not be waiting for a magical coder fairy to pop out of the woodworks and make demos not lag just as soon as they're home from the corner store for smokes and milk

## Changelog
:cl:
admin: demos are gone, they were super duper laggy for little gain
/:cl:

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -546,8 +546,4 @@
 /turf/bullet_act(obj/projectile/P)
 	. = ..()
 	if(. != BULLET_ACT_FORCE_PIERCE)
-		. =  BULLET_ACT_TURF
-
-/turf/setDir()
-	. = ..()
-	SSdemo.mark_turf(src)
+		. =  BULLET_ACT_TURF
```
Comment (Ghilker) [inline code/game/turfs/turf.dm:549]: newline here

**Comment by Dorsisdwarf** [comment]
damn, these were incredibly nice. but they are pretty clunky

**Comment by Tlaltecuhtli** [comment]
cant you just make demos record 1 min before someone gets murdered?

**Comment by BeeSting12** [comment]
these were cool but making the admins job easier for such a performance loss is dumb

**Comment by Cyberboss** [comment]
This was already made a config option (#50615) and I'm pretty sure it's disabled now.

**Comment by Iamgoofball** [comment]
Excuse me? We're not maintaining the code. It needs to go, it's going to stay disabled forever.

**Comment by Neo-0** [comment]
https://github.com/tgstation/tgstation/pull/50653 oof

**Comment by optimumtact** [comment]
We still need to remove it as action indicated they didn't want to go further with it.

**Comment by Moja10** [comment]
Just optimize the code 4head

**Comment by ExcessiveUseOfCobblestone** [comment]
wyoi

**Comment by actioninja** [comment]
To elaborate a bit on why I'm not super interested in supporting it, it's not particularly poorly made it just does so much. I don't know if this style of replay recording is ever going to work in native dm. dm is just too slow on simple operations and too much stuff changes that's involved.
On top of that the only reason why I went forward with porting it was I was under the impression it ran fine on yogs because they had previously said it ran fine. After it was merged I got asked by yogs coders why we went ahead with it when it was so laggy. Turns out it did run like total ass they just dealt with it.

**Pull request merged**